insights/parsers/tests/test_samba_smb_conf.py

```diff
@@ -0,0 +1,245 @@
+    comment = All Printers
+    path = /var/spool/samba
+    browseable = no
+    guest ok = no
+    writable = no
+    printable = yes
+
+# Un-comment the following and create the netlogon directory for Domain Logons
+;   [netlogon]
+;   comment = Network Logon Service
+;   path = /var/lib/samba/netlogon
+;   guest ok = yes
+;   writable = no
+;   share modes = no
+
+
+# Un-comment the following to provide a specific roving profile share
+# the default is to use the user's home directory
+;   [Profiles]
+;   path = /var/lib/samba/profiles
+;   browseable = no
+;   guest ok = yes
+
+
+# A publicly accessible directory, but read only, except for people in
+# the "staff" group
+;   [public]
+;   comment = Public Stuff
+;   path = /home/samba
+;   public = yes
+;   writable = yes
+;   printable = no
+;   write list = +staff
+
+[ GlObAl  ]
+
+# Samba also automatically treats non-lowercase section names as lowercase and strips whitespace.
+# This behavior can be checked with `testparm` again.
+this option should also be in global = true
+
+[ GlObAl  ]
+
+# This tests specifically that two same-named sections are automatically merged in case the
+# RawConfigParser's behavior ever changes.
+this another option should also be in global = 1
+"""
+
+# This is the output of testparm -s parsing the above SAMBA_CONFIG
+SAMBA_TESTPARM_OUTPUT = '''
+# Global parameters
+[global]
+        log file = /var/log/samba/log.%m
+        max log size = 50
+        security = USER
+        server string = Samba Server Version %v
+        workgroup = MYGROUP
+        idmap config * : backend = tdb
+        cups options = raw
+
+
+[homes]
+        browseable = No
+        comment = Home Directories
+        read only = No
+
+
+[printers]
+        browseable = No
+        comment = All Printers
+        path = /var/spool/samba
+        printable = Yes
+'''
+
+
+def test_match():
+    config = samba_smb_conf.SambaConfig(context_wrap(SAMBA_TESTPARM_OUTPUT))
+
+    assert config.get('global', 'workgroup') == 'MYGROUP'
+    assert config.get('global', 'server string') == 'Samba Server Version %v'
+    assert not config.has_option('global', 'netbios name')
+    assert config.get('global', 'log file') == '/var/log/samba/log.%m'
+    assert config.get('global', 'max log size') == '50'
+
+    assert config.get('global', 'security') == 'USER'
+
+    assert config.get('global', 'cups options') == 'raw'
+
+    assert config.get('homes', 'comment') == 'Home Directories'
+    assert config.get('homes', 'browseable') == 'No'
+    assert config.get('homes', 'read only') == 'No'
+    assert not config.has_option('homes', 'valid users')
+
+    assert config.get('printers', 'comment') == 'All Printers'
+    assert config.get('printers', 'path') == '/var/spool/samba'
+    assert config.get('printers', 'browseable') == 'No'
+    assert config.get('printers', 'printable') == 'Yes'
+
+    assert 'netlogin' not in config
+    assert 'Profiles' not in config
+    assert 'public' not in config
```

Comment:
Please add testing for your parser docstring examples.  Here's an [example](https://github.com/RedHatInsights/insights-core/blob/f3b02e09a8784b9e3d0618d2319b9c34f457c16c/insights/parsers/tests/test_rhsm_releasever.py#L45) of how that can be done.

Reply:
We have a test, I renamed it and move it to the end now.